R/metadata.R

```diff
@@ -120,6 +120,79 @@ static_orderly_artefact <- function(args) {
 }
 
 
+##' Declare a dependency on another packet
+##'
+##' @title Declare a dependency
+##'
+##' @param name The name of the packet to depend on
+##'
+##' @param query The query to search for; often this will simply be
+##'   the string `latest`, indicating the most recent version. You may
+##'   want a more complex query here though.
+##'
+##' @param use A named character vector of filenames to copy from the
+##'   upstream packet. The name corresponds to the destination name,
+##'   so c(here.csv = "there.csv") will take the upstream file
+##'   `there.csv` and copy it over as `here.csv`.
+##'
+##' @return Undefined
+##' @export
+orderly_depends <- function(name, query, use) {
```

Comment:
Feels like it would be nice if you could just do `function(name, query, ...)` where `...` is the named file arguments.

Reply:
I agree, but here's the pair of issues issue with that:

First, In the case where you want to compute the filenames passing in dots is hard as R does not have a spread/splat operator. This will be an issue for Pete's case where we are pulling in things from different regions and renaming them as what we can do with an explicit `use` argument is:

```r
for (r in regions) {
  query <- rlang::expr(parameter:region == !!r)
  use <- setNames("data.rds", file.path("data", paste0(r, ".rds")))
  orderly_depends("data", query, use)
}
```

Second, you can compromise by adding a get collisions with names, so we could write:

```
orderly_depends <- function(name, query, ..., use = c(...)) {
}
```

or by carefully processing `...`, but this requires some care with any analysis of the contents in `static_orderly_dependency` I think.

I had a look at sorting this out but it looks complicated enough that I'd rather kick it down the road a bit

Reply:
https://mrc-ide.myjetbrains.com/youtrack/issue/mrc-4173